### PR TITLE
JAMES-4032 Allow relaxed sender verification

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-2268/2/

Related

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
FYI the CI didn't get triggered after last fixup push... I started it manually. To see if CI is going nuts again or not

File: protocols/smtp/src/main/java/org/apache/james/protocols/smtp/SMTPConfiguration.java [hunk 1]
```diff
@@ -36,6 +37,21 @@
...
 public interface SMTPConfiguration extends ProtocolConfiguration {
+    enum SenderVerificationMode {
+        STRICT,
+        RELAXED,
+        DISABLED;
+
+        // TODO unit tests
+        public static SenderVerificationMode parse(String value) {
+            return switch (value.toLowerCase(Locale.US).trim()) {
+                case "true", "strict" -> STRICT;
```
Comment (vttranlina) [inline protocols/smtp/src/main/java/org/apache/james/protocols/smtp/SMTPConfiguration.java:48]: maybe you will like util `org.apache.commons.lang3.BooleanUtils#toBoolean(java.lang.String)`

Comment (chibenwa) [reply protocols/smtp/src/main/java/org/apache/james/protocols/smtp/SMTPConfiguration.java:48]: Thanks for sharing.

It do not handle strict / disabled.

**Pull request merged**